### PR TITLE
Upgrade Eureka

### PR DESCRIPTION
**Pull request by jkschneider**
* Removes Servo from EurekaHealthIndicator
* Removes DataCenterInfo implementation and custom codec
* To customize the id of `InstanceInfo` set ~~`eureka.instance.sid`~~ `eureka.instance.instanceId`

### CODE REVIEW

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientConfigBean.java [hunk 1]
```diff
@@ -123,6 +124,12 @@
 
 	private boolean onDemandUpdateStatusChange = true;
 
+    private String encoderName;
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientConfigBean.java]: We use tabs, not spaces :-)

**Comment by spencergibb** [comment]
Did you see [this comment](https://github.com/Netflix/eureka/pull/649#issuecomment-142772696) by @qiangdavidliu? Will that change come in 1.2.5-rc.2?

**Comment by qiangdavidliu** [comment]
Yes should be there by rc.2

**Comment by spencergibb** [comment]
@qiangdavidliu thanks!

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientConfigBean.java (hunk 1 above)
Comment (jkschneider) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientConfigBean.java]: argg... Eclipse plugin in IntelliJ could use improvement :)

**Comment by jkschneider** [comment]
Reopen

**Comment by spencergibb** [comment]
I can confirm this works by setting `eureka.instance.sid`.
eurekawithsid

**Comment by spencergibb** [comment]
Polished and merged via 8f9acab3bdbebfcf5687ec601904ee4f718ae25d.

Most notable `EurekaInstanceConfigBean.getSID` falls back to `EurekaInstanceConfigBean.metadataMap.instanceId` for
backwards compatibility.

**Pull request closed without merging**